microsoft/atlas-design
# Migrate from Yarn to Npm

Task: [task-414149](https://dev.azure.com/ceapex/Engineering/_boards/board/t/Fox%20Team/Stories/?workitem=414149)

Link: [preview-111](https://design.docs.microsoft.com/pulls/111)

This PR migrates from `yarn` to Node 15.7.0 and npm 7.4.0 to match the recent work in Docs-UI.

## Testing

1. Clone a new version of Atlas Design as a clean slate. (Don't mess up your primary environment.) Use the following command in a folder of your choice to clone the project into a directory called "atlas-design-npm": `git clone https://github.com/microsoft/atlas-design.git --single-branch atlas-design-npm`
2. Install version 15.7.0 of Node. Do one of the following:
- Install using the [Windows Installer from nodejs.org](https://nodejs.org/dist/v15.7.0/node-v15.7.0-x64.msi) . To revert, you will have to uninstall Node from Control Panel.
- Install version 15.7.0 using the Node Version Manager (NVM)  using the commands nvm install 15.7.0 and nvm use 15.7.0. (To switch back, use nvm install 14.8.0 and nvm use 14.8.0. To see available versions installed, use nvm list.)
3. In the repo, run `npm install`. Dependencies should install without issues.
4. Test the main commands we use in the root directory. 
- npm run lint
- npm run lint:fix
- npm run start
- `npm run serve:site` (same as npm run start)
- npm run serve:css
- npm run build:css
- npm run build:site
- npm run build:tokens
- npm run toc
- `npm run prestart` (same as npm run build:tokens)
- `npm run prebuild:site` (same as npm run build:tokens)

**- npm run release  (can't test until we want to publish?)**

### Success Criteria
- Local development builds should pass without problems, and the workflow should not be affected.
- CI builds should pass without any problems.
